core/ecschema-editing/src/Editing/Properties.ts

```diff
@@ -0,0 +1,298 @@
+import { ArrayProperty, ECClass, ECName, ECObjectsError, ECObjectsStatus, EnumerationProperty, PrimitiveProperty, SchemaItemKey, SchemaItemType } from "@itwin/ecschema-metadata";
+import { SchemaContextEditor } from "./Editor";
+import { MutableArrayProperty } from "./Mutable/MutableArrayProperty";
+import { MutableProperty } from "./Mutable/MutableProperty";
+import { MutablePrimitiveOrEnumPropertyBase } from "./Mutable/MutablePrimitiveOrEnumProperty";
+import { MutableClass } from "./Mutable/MutableClass";
+
+type MutablePropertyType = MutableProperty | MutableArrayProperty | MutablePrimitiveOrEnumPropertyBase;
+
+/**
+ * @alpha
+ * A class allowing editing of attributes of the base Property class.
+ */
+export class Properties {
```

Comment:
With introducing the Properties editor, we now have the several places where callers can edit certain property aspects. For instance the _PropertyCategories_ can be set in _ECClasse_s only, the property name can be set here and in ECClasses as well, ...

In my opinion, we should probably move all property related stuff from the ECClass editor into the properties editor as a new central place for everything property related as long all of this is still _alpha_.

I'm aware that this might break the merging API callers, but that should be easy to adopt.

Reply:
Yeah, I thought the same.  Wasn't sure if it should be in this PR, but I will go ahead and move the API.

Reply:
My latest push has these changes.